### ISSUE TITLE
Can't build using 1.4.0 version of module

### ISSUE THREAD

**Issue by jwogan5** [hansemannn/titanium-firebase-analytics#28]
I am using Appc SDK 7.5.0.   With Firebase core 2.0.0 and an older version of this module, everything works fine.  ( As you can see I am using the newest firebase core ).

• When I try to build to a device I get an error about "DTDeviceKit: deviceType from 00008006-000668380CE2002E was NULL"

• When I try to build to a simulator I get errors about "Undefined symbols for architecture x86_64"

Error log for build against simulator:

[TRACE]  Undefined symbols for architecture x86_64:
[TRACE]    "_OBJC_CLASS_$_APMAnalytics", referenced from:
[TRACE]        objc-class-ref in FirebaseAnalytics(FIRAnalytics_ca9cdf99cfbae2f51d040b73dd242e94.o)
[TRACE]    "_kAPMMonitorLogTagOptionKey", referenced from:
[TRACE]        +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics(FIRAnalytics_ca9cdf99cfbae2f51d040b73dd242e94.o)
[TRACE]    "_gAPMAppMeasurementLibraryVersion", referenced from:
[TRACE]        +[FIRAnalytics initialize] in FirebaseAnalytics(FIRAnalytics_ca9cdf99cfbae2f51d040b73dd242e94.o)
[TRACE]    "_kAPMIsAnalyticsCollectionEnabled", referenced from:
[TRACE]        +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics(FIRAnalytics_ca9cdf99cfbae2f51d040b73dd242e94.o)
[TRACE]    "_OBJC_METACLASS_$_APMMeasurement", referenced from:
[TRACE]        _OBJC_METACLASS_$_FIRAMeasurement in FirebaseAnalytics(FIRAMeasurement_7113499ff6c634053dac5bf06844d505.o)
[TRACE]    "_APMFormattedUserPropertyName", referenced from:
[TRACE]        +[FIRAnalytics setUserPropertyString:forName:] in FirebaseAnalytics(FIRAnalytics_ca9cdf99cfbae2f51d040b73dd242e94.o)
[TRACE]    "_OBJC_METACLASS_$_APMSessionReporter", referenced from:
[TRACE]        _OBJC_METACLASS_$_FIRASessionReporter in FirebaseAnalytics(FIRASessionReporter_fedb19668cf4c09fe19f4dd3dd423569.o)
[TRACE]    "_OBJC_CLASS_$_APMValue", referenced from:
[TRACE]        _OBJC_CLASS_$_FIRAValue in FirebaseAnalytics(FIRAValue_2e0ad7f420162fa95ff8e12af195aa2b.o)
[TRACE]    "_OBJC_CLASS_$_APMMeasurement", referenced from:
[TRACE]        objc-class-ref in FirebaseAnalytics(FIRAMeasurement_7113499ff6c634053dac5bf06844d505.o)
[TRACE]        _OBJC_CLASS_$_FIRAMeasurement in FirebaseAnalytics(FIRAMeasurement_7113499ff6c634053dac5bf06844d505.o)
[TRACE]        objc-class-ref in FirebaseAnalytics(FIRAnalytics_ca9cdf99cfbae2f51d040b73dd242e94.o)
[TRACE]    "_OBJC_METACLASS_$_APMConditionalUserProperty", referenced from:
[TRACE]        _OBJC_METACLASS_$_FIRAConditionalUserProperty in FirebaseAnalytics(FIRAConditionalUserProperty_991dd1dfefbe571de6dbf327969af9a0.o)
[TRACE]    "_OBJC_METACLASS_$_APMAdExposureReporter", referenced from:
[TRACE]        _OBJC_METACLASS_$_FIRAAdExposureReporter in FirebaseAnalytics(FIRAAdExposureReporter_9f9da43a0e4ed4258a1c1de8d08084f3.o)
[TRACE]    "_OBJC_METACLASS_$_APMScreenViewReporter", referenced from:
[TRACE]        _OBJC_METACLASS_$_FIRAScreenViewReporter in FirebaseAnalytics(FIRAScreenViewReporter_ded399434f21648531c172048caa1206.o)
[TRACE]    "_OBJC_CLASS_$_APMScreenViewReporter", referenced from:
[TRACE]        objc-class-ref in FirebaseAnalytics(FIRAScreenViewReporter_ded399434f21648531c172048caa1206.o)
[TRACE]        _OBJC_CLASS_$_FIRAScreenViewReporter in FirebaseAnalytics(FIRAScreenViewReporter_ded399434f21648531c172048caa1206.o)
[TRACE]    "_OBJC_CLASS_$_APMAdExposureReporter", referenced from:
[TRACE]        _OBJC_CLASS_$_FIRAAdExposureReporter in FirebaseAnalytics(FIRAAdExposureReporter_9f9da43a0e4ed4258a1c1de8d08084f3.o)
[TRACE]    "_APMFormattedEventName", referenced from:
[TRACE]        +[FIRAnalytics logEventWithOrigin:name:parameters:] in FirebaseAnalytics(FIRAnalytics_ca9cdf99cfbae2f51d040b73dd242e94.o)
[TRACE]    "_OBJC_CLASS_$_APMConditionalUserProperty", referenced from:
[TRACE]        _OBJC_CLASS_$_FIRAConditionalUserProperty in FirebaseAnalytics(FIRAConditionalUserProperty_991dd1dfefbe571de6dbf327969af9a0.o)
[TRACE]    "_OBJC_CLASS_$_APMEvent", referenced from:
[TRACE]        _OBJC_CLASS_$_FIRAEvent in FirebaseAnalytics(FIRAEvent_c32f39c4e1681f460e696def59992d96.o)
[TRACE]    "_kAPMSafelistedEventsOptionKey", referenced from:
[TRACE]        +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics(FIRAnalytics_ca9cdf99cfbae2f51d040b73dd242e94.o)
[TRACE]    "_OBJC_METACLASS_$_APMEvent", referenced from:
[TRACE]        _OBJC_METACLASS_$_FIRAEvent in FirebaseAnalytics(FIRAEvent_c32f39c4e1681f460e696def59992d96.o)
[TRACE]    "_kAPMIsAnalyticsCollectionDeactivated", referenced from:
[TRACE]        +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics(FIRAnalytics_ca9cdf99cfbae2f51d040b73dd242e94.o)
[TRACE]    "_OBJC_CLASS_$_APMConditionalUserPropertyController", referenced from:
[TRACE]        _OBJC_CLASS_$_FIRAConditionalUserPropertyController in FirebaseAnalytics(FIRAConditionalUserPropertyController_faf82aa1c60b80614f617fe3e96f08fe.o)
[TRACE]    "_OBJC_CLASS_$_APMSessionReporter", referenced from:
[TRACE]        _OBJC_CLASS_$_FIRASessionReporter in FirebaseAnalytics(FIRASessionReporter_fedb19668cf4c09fe19f4dd3dd423569.o)
[TRACE]    "_OBJC_METACLASS_$_APMIdentifiers", referenced from:
[TRACE]        _OBJC_METACLASS_$_FIRAIdentifiers in FirebaseAnalytics(FIRAIdentifiers_16c07d77c69044b43f7b9de70da25f9f.o)
[TRACE]    "_OBJC_METACLASS_$_APMUserAttribute", referenced from:
[TRACE]        _OBJC_METACLASS_$_FIRAUserAttribute in FirebaseAnalytics(FIRAUserAttribute_66224d31710f67eef7b8c0e9f1c05115.o)
[TRACE]    "_kAPMAnalyticsConfiguration", referenced from:
[TRACE]        +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics(FIRAnalytics_ca9cdf99cfbae2f51d040b73dd242e94.o)
[TRACE]    "_kAPMAppMeasurementOriginFirebase", referenced from:
[TRACE]        +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics(FIRAnalytics_ca9cdf99cfbae2f51d040b73dd242e94.o)
[TRACE]    "_OBJC_METACLASS_$_APMValue", referenced from:
[TRACE]        _OBJC_METACLASS_$_FIRAValue in FirebaseAnalytics(FIRAValue_2e0ad7f420162fa95ff8e12af195aa2b.o)
[TRACE]    "_OBJC_CLASS_$_APMIdentifiers", referenced from:
[TRACE]        _OBJC_CLASS_$_FIRAIdentifiers in FirebaseAnalytics(FIRAIdentifiers_16c07d77c69044b43f7b9de70da25f9f.o)
[TRACE]    "_OBJC_METACLASS_$_APMConditionalUserPropertyController", referenced from:
[TRACE]        _OBJC_METACLASS_$_FIRAConditionalUserPropertyController in FirebaseAnalytics(FIRAConditionalUserPropertyController_faf82aa1c60b80614f617fe3e96f08fe.o)
[TRACE]    "_OBJC_CLASS_$_APMUserAttribute", referenced from:
[TRACE]        _OBJC_CLASS_$_FIRAUserAttribute in FirebaseAnalytics(FIRAUserAttribute_66224d31710f67eef7b8c0e9f1c05115.o)
[TRACE]    "_OBJC_CLASS_$_FIRComponentContainer", referenced from:
[TRACE]        objc-class-ref in FIRAnalyticsConnector(FIRAnalyticsConnector_83e864ed9fa8f98ebe9f4740c2eef8b8.o)
[TRACE]    "_OBJC_CLASS_$_FIRComponent", referenced from:
[TRACE]        objc-class-ref in FIRAnalyticsConnector(FIRAnalyticsConnector_83e864ed9fa8f98ebe9f4740c2eef8b8.o)
[TRACE]  ld: symbol(s) not found for architecture x86_64
[TRACE]  clang: error
[TRACE]  : linker command failed with exit code 1 (use -v to see invocation)
[ERROR]  ** BUILD FAILED **

Error log for build against device:

[TRACE]  CompileC /Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/build/Intermediates/Dealer\ Center.build/Debug-iphoneos/Dealer\ Center.build/Objects-normal/arm64/TiUIiOSMenuPopupProxy.o /Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/Classes/TiUIiOSMenuPopupProxy.m normal arm64 objective-c com.apple.compilers.llvm.clang.1_0.compiler (in target: Dealer Center)
[TRACE]      cd /Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone
[TRACE]      export LANG=en_US.US-ASCII
[TRACE]      /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x objective-c -arch arm64 -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -std=c99 -fmodules -gmodules -fmodules-cache-path=/Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/DerivedData/ModuleCache.noindex -fmodules-prune-interval=86400 -fmodules-prune-after=345600 -fbuild-session-file=/Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/DerivedData/ModuleCache.noindex/Session.modulevalidation -fmodules-validate-once-per-build-session -Wnon-modular-include-in-framework-module -Werror=non-modular-include-in-framework-module -Wno-trigraphs -fpascal-strings -O0 -fno-common -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wunreachable-code -Wno-implicit-atomic-properties -Wno-objc-interface-ivars -Wno-arc-repeated-use-of-weak -Wduplicate-method-match -Wno-missing-braces -Wno-parentheses -Wswitch -Wunused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wno-unused-value -Wempty-body -Wuninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wconstant-conversion -Wint-conversion -Wbool-conversion -Wenum-conversion -Wno-float-conversion -Wnon-literal-null-conversion -Wobjc-literal-conversion -Wshorten-64-to-32 -Wpointer-sign -Wno-newline-eof -Wno-selector -Wno-strict-selector-match -Wno-undeclared-selector -Wno-deprecated-implementations -DDEPLOYTYPE=development -DLAUNCHSCREEN_STORYBOARD=1 -DDEFAULT_BGCOLOR_RED=1 -DDEFAULT_BGCOLOR_GREEN=1 -DDEFAULT_BGCOLOR_BLUE=1 -DTI_LOG_SERVER_PORT=45296 -DOBJC_OLD_DISPATCH_PROTOTYPES=0 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.1.sdk -fstrict-aliasing -Wprotocol -Wdeprecated-declarations -miphoneos-version-min=9.0 -g -Wno-sign-conversion -Winfinite-recursion -Wcomma -Wblock-capture-autoreleasing -Wno-strict-prototypes -Wno-semicolon-before-method-body -Wno-unguarded-availability -index-store-path /Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/DerivedData/Index/DataStore -iquote /Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/build/Intermediates/Dealer\ Center.build/Debug-iphoneos/Dealer\ Center.build/Dealer\ Center-generated-files.hmap -I/Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/build/Intermediates/Dealer\ Center.build/Debug-iphoneos/Dealer\ Center.build/Dealer\ Center-own-target-headers.hmap -I/Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/build/Intermediates/Dealer\ Center.build/Debug-iphoneos/Dealer\ Center.build/Dealer\ Center-all-target-headers.hmap -iquote /Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/build/Intermediates/Dealer\ Center.build/Debug-iphoneos/Dealer\ Center.build/Dealer\ Center-project-headers.hmap -I/Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/build/Products/Debug-iphoneos/include -I/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.1.sdk/usr/include/libxml2 -Iheaders -I/Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/build/Intermediates/Dealer\ Center.build/Debug-iphoneos/Dealer\ Center.build/DerivedSources/arm64 -I/Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/build/Intermediates/Dealer\ Center.build/Debug-iphoneos/Dealer\ Center.build/DerivedSources -Wno-arc-performSelector-leaks -F/Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/build/Products/Debug-iphoneos -F/Users/jason.wogan/Library/Application\ Support/Titanium/modules/iphone/firebase.core/2.0.0/platform -F/Users/jason.wogan/Library/Application\ Support/Titanium/modules/iphone/com.traderinteractive.appsee/0.0.1/platform -F/Users/jason.wogan/Library/Application\ Support/Titanium/modules/iphone/com.traderinteractive.fyusion/0.0.2/platform -F/Users/jason.wogan/Library/Application\ Support/Titanium/modules/iphone/firebase.cloudmessaging/2.0.1/platform -F/Users/jason.wogan/Library/Application\ Support/Titanium/modules/iphone/firebase.analytics/1.3.0/platform -F/Users/jason.wogan/Library/Application\ Support/Titanium/modules/iphone/ti.crashlytics/1.1.1/platform -DDEBUG -include /Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/build/Intermediates/PrecompiledHeaders/SharedPrecompiledHeaders/11076202699044268272/Dealer_Center_Prefix.pch -MMD -MT dependencies -MF /Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/build/Intermediates/Dealer\ Center.build/Debug-iphoneos/Dealer\ Center.build/Objects-normal/arm64/TiUIiOSMenuPopupProxy.d --serialize-diagnostics /Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/build/Intermediates/Dealer\ Center.build/Debug-iphoneos/Dealer\ Center.build/Objects-normal/arm64/TiUIiOSMenuPopupProxy.dia -c /Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/Classes/TiUIiOSMenuPopupProxy.m -o /Users/jason.wogan/work/mobileapps/mobile-imt-v2/build/iphone/build/Intermediates/Dealer\ Center.build/Debug-iphoneos/Dealer\ Center.build/Objects-normal/arm64/TiUIiOSMenuPopupProxy.o
[ERROR]  2018-11-26 15:54:12.250 xcodebuild[17594:1076757]  DTDeviceKit: deviceType from 00008006-000668380CE2002E was NULL

**Comment by gareth-poole** [comment]
I'm having the same problem

**Comment by hansemannn** [comment]
It looks like a module collision between APM and Firebase. APM seems to use the same classes than Firebase. Please make sure to report that so it can be fixed on the APM side. We are using 1.4.0 in production without issues.

**Comment by Astrovic** [comment]
I've the same issue. Just using this module:

```
<modules>
      <module>hyperloop</module>
      <module platform="iphone" version="2.0.0">firebase.core</module>
      <module platform="iphone" version="1.4.0">firebase.analytics</module>
</modules>
```

the build failed with this error:

```
[TRACE] Undefined symbols for architecture x86_64:
[TRACE]   "_OBJC_CLASS_$_APMAnalytics", referenced from:
[TRACE]       objc-class-ref in FirebaseAnalytics(FIRAnalytics_ca9cdf99cfbae2f51d040b73dd242e94.o)
[TRACE]   "_kAPMMonitorLogTagOptionKey", referenced from:
[TRACE]       +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics(FIRAnalytics_ca9cdf99cfbae2f51d040b73dd242e94.o)
[TRACE]   "_gAPMAppMeasurementLibraryVersion", referenced from:
[TRACE]       +[FIRAnalytics initialize] in FirebaseAnalytics(FIRAnalytics_ca9cdf99cfbae2f51d040b73dd242e94.o)
[TRACE]   "_kAPMIsAnalyticsCollectionEnabled", referenced from:
[TRACE]       +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics(FIRAnalytics_ca9cdf99cfbae2f51d040b73dd242e94.o)
[TRACE]   "_OBJC_METACLASS_$_APMMeasurement", referenced from:
[TRACE]       _OBJC_METACLASS_$_FIRAMeasurement in FirebaseAnalytics(FIRAMeasurement_7113499ff6c634053dac5bf06844d505.o)
[TRACE]   "_APMFormattedUserPropertyName", referenced from:
[TRACE]       +[FIRAnalytics setUserPropertyString:forName:] in FirebaseAnalytics(FIRAnalytics_ca9cdf99cfbae2f51d040b73dd242e94.o)
[TRACE]   "_OBJC_METACLASS_$_APMSessionReporter", referenced from:
[TRACE]       _OBJC_METACLASS_$_FIRASessionReporter in FirebaseAnalytics(FIRASessionReporter_fedb19668cf4c09fe19f4dd3dd423569.o)
[TRACE]   "_OBJC_CLASS_$_APMValue", referenced from:
[TRACE]       _OBJC_CLASS_$_FIRAValue in FirebaseAnalytics(FIRAValue_2e0ad7f420162fa95ff8e12af195aa2b.o)
[TRACE]   "_OBJC_CLASS_$_APMMeasurement", referenced from:
[TRACE]       objc-class-ref in FirebaseAnalytics(FIRAMeasurement_7113499ff6c634053dac5bf06844d505.o)
[TRACE]       _OBJC_CLASS_$_FIRAMeasurement in FirebaseAnalytics(FIRAMeasurement_7113499ff6c634053dac5bf06844d505.o)
[TRACE]       objc-class-ref in FirebaseAnalytics(FIRAnalytics_ca9cdf99cfbae2f51d040b73dd242e94.o)
[TRACE]   "_OBJC_METACLASS_$_APMConditionalUserProperty", referenced from:
[TRACE]       _OBJC_METACLASS_$_FIRAConditionalUserProperty in FirebaseAnalytics(FIRAConditionalUserProperty_991dd1dfefbe571de6dbf327969af9a0.o)
[TRACE]   "_OBJC_METACLASS_$_APMAdExposureReporter", referenced from:
[TRACE]       _OBJC_METACLASS_$_FIRAAdExposureReporter in FirebaseAnalytics(FIRAAdExposureReporter_9f9da43a0e4ed4258a1c1de8d08084f3.o)
[TRACE]   "_OBJC_METACLASS_$_APMScreenViewReporter", referenced from:
[TRACE]       _OBJC_METACLASS_$_FIRAScreenViewReporter in FirebaseAnalytics(FIRAScreenViewReporter_ded399434f21648531c172048caa1206.o)
[TRACE]   "_OBJC_CLASS_$_APMScreenViewReporter", referenced from:
[TRACE]       objc-class-ref in FirebaseAnalytics(FIRAScreenViewReporter_ded399434f21648531c172048caa1206.o)
[TRACE]       _OBJC_CLASS_$_FIRAScreenViewReporter in FirebaseAnalytics(FIRAScreenViewReporter_ded399434f21648531c172048caa1206.o)
[TRACE]   "_OBJC_CLASS_$_APMAdExposureReporter", referenced from:
[TRACE]       _OBJC_CLASS_$_FIRAAdExposureReporter in FirebaseAnalytics(FIRAAdExposureReporter_9f9da43a0e4ed4258a1c1de8d08084f3.o)
[TRACE]   "_APMFormattedEventName", referenced from:
[TRACE]       +[FIRAnalytics logEventWithOrigin:name:parameters:] in FirebaseAnalytics(FIRAnalytics_ca9cdf99cfbae2f51d040b73dd242e94.o)
[TRACE]   "_OBJC_CLASS_$_APMConditionalUserProperty", referenced from:
[TRACE]       _OBJC_CLASS_$_FIRAConditionalUserProperty in FirebaseAnalytics(FIRAConditionalUserProperty_991dd1dfefbe571de6dbf327969af9a0.o)
[TRACE]   "_OBJC_CLASS_$_APMEvent", referenced from:
[TRACE]       _OBJC_CLASS_$_FIRAEvent in FirebaseAnalytics(FIRAEvent_c32f39c4e1681f460e696def59992d96.o)
[TRACE]   "_kAPMSafelistedEventsOptionKey", referenced from:
[TRACE]       +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics(FIRAnalytics_ca9cdf99cfbae2f51d040b73dd242e94.o)
[TRACE]   "_OBJC_METACLASS_$_APMEvent", referenced from:
[TRACE]       _OBJC_METACLASS_$_FIRAEvent in FirebaseAnalytics(FIRAEvent_c32f39c4e1681f460e696def59992d96.o)
[TRACE]   "_kAPMIsAnalyticsCollectionDeactivated", referenced from:
[TRACE]       +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics(FIRAnalytics_ca9cdf99cfbae2f51d040b73dd242e94.o)
[TRACE]   "_OBJC_CLASS_$_APMConditionalUserPropertyController", referenced from:
[TRACE]       _OBJC_CLASS_$_FIRAConditionalUserPropertyController in FirebaseAnalytics(FIRAConditionalUserPropertyController_faf82aa1c60b80614f617fe3e96f08fe.o)
[TRACE]   "_OBJC_CLASS_$_APMSessionReporter", referenced from:
[TRACE]       _OBJC_CLASS_$_FIRASessionReporter in FirebaseAnalytics(FIRASessionReporter_fedb19668cf4c09fe19f4dd3dd423569.o)
[TRACE]   "_OBJC_METACLASS_$_APMIdentifiers", referenced from:
[TRACE]       _OBJC_METACLASS_$_FIRAIdentifiers in FirebaseAnalytics(FIRAIdentifiers_16c07d77c69044b43f7b9de70da25f9f.o)
[TRACE]   "_OBJC_METACLASS_$_APMUserAttribute", referenced from:
[TRACE]       _OBJC_METACLASS_$_FIRAUserAttribute in FirebaseAnalytics(FIRAUserAttribute_66224d31710f67eef7b8c0e9f1c05115.o)
[TRACE]   "_kAPMAnalyticsConfiguration", referenced from:
[TRACE]       +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics(FIRAnalytics_ca9cdf99cfbae2f51d040b73dd242e94.o)
[TRACE]   "_kAPMAppMeasurementOriginFirebase", referenced from:
[TRACE]       +[FIRAnalytics startWithConfiguration:options:] in FirebaseAnalytics(FIRAnalytics_ca9cdf99cfbae2f51d040b73dd242e94.o)
[TRACE]   "_OBJC_METACLASS_$_APMValue", referenced from:
[TRACE]       _OBJC_METACLASS_$_FIRAValue in FirebaseAnalytics(FIRAValue_2e0ad7f420162fa95ff8e12af195aa2b.o)
[TRACE]   "_OBJC_CLASS_$_APMIdentifiers", referenced from:
[TRACE]       _OBJC_CLASS_$_FIRAIdentifiers in FirebaseAnalytics(FIRAIdentifiers_16c07d77c69044b43f7b9de70da25f9f.o)
[TRACE]   "_OBJC_METACLASS_$_APMConditionalUserPropertyController", referenced from:
[TRACE]       _OBJC_METACLASS_$_FIRAConditionalUserPropertyController in FirebaseAnalytics(FIRAConditionalUserPropertyController_faf82aa1c60b80614f617fe3e96f08fe.o)
[TRACE]   "_OBJC_CLASS_$_APMUserAttribute", referenced from:
[TRACE]       _OBJC_CLASS_$_FIRAUserAttribute in FirebaseAnalytics(FIRAUserAttribute_66224d31710f67eef7b8c0e9f1c05115.o)
[TRACE]   "_OBJC_CLASS_$_FIRComponentContainer", referenced from:
[TRACE]       objc-class-ref in FIRAnalyticsConnector(FIRAnalyticsConnector_83e864ed9fa8f98ebe9f4740c2eef8b8.o)
[TRACE]   "_OBJC_CLASS_$_FIRComponent", referenced from:
[TRACE]       objc-class-ref in FIRAnalyticsConnector(FIRAnalyticsConnector_83e864ed9fa8f98ebe9f4740c2eef8b8.o)
[TRACE] ld: symbol(s) not found for architecture x86_64
[TRACE] clang
[TRACE] : error: linker command failed with exit code 1 (use -v to see invocation)
[ERROR] ** BUILD FAILED **
[ERROR] The following build commands failed:
[ERROR]         Ld /Users/Astrovic/Dati/Applicazioni/FA/build/iphone/build/Products/Debug-iphonesimulator/FA.app/FA normal x86_64
[ERROR] (1 failure)
```

**Comment by jwogan5** [comment]
So I actually got the new versions of firebase analytics and firebase cloud messaging modules working today.  Basically, the latest release version of firebase core (2.0.0) is not the latest one that works with the new modules.  The new modules require firebase core ( 2.1.0 ).  I cloned the firebase core code and then built it myself which built the 2.1.0 module. I also added x86_64 and i386 to all of the arch versions within Xcode for all 3 modules. Once I built all 3 modules myself with the latest code from master my app built successfully to both the simulator and my device.

If I come across any other issues I will update this issue.
